openingd/openingd.c

```diff
@@ -1537,8 +1537,13 @@ static u8 *handle_peer_in(struct state *state)
 	case WIRE_REPLY_CHANNEL_RANGE:
 	case WIRE_GOSSIP_TIMESTAMP_FILTER:
 	case WIRE_ERROR:
-	case WIRE_CHANNEL_REESTABLISH:
+		if (handle_peer_gossip_or_error(state->pps,
+						&state->channel_id, msg))
+			return NULL;
+		break;
+
 	/* These are all protocol violations at this stage. */
+	case WIRE_CHANNEL_REESTABLISH:
```

Comment:
Hmm, I had to read the code, but this is wrong. handle_peer_gossip_or_error also handles messages (esp. reestablish) which are talking about the wrong channel, and sends a more useful error message.

Reply:
Thanks my mistake, I should've read the function description more clearly.